### PR TITLE
Image

### PR DESCRIPTION
**Pull request by moses369**
ShareBtn Component Changes
- tracking mouse position was useless, only needed title attribute
-Implemented links into share btn

Redux Changes
- Replaced image-slice for prod-info reducing the need for unnecessary complexity

### CODE REVIEW

**Review by EntropicWind — DISMISSED** [review]

File: components/ProductInfo/Cart.js [hunk 1]
```diff
@@ -38,42 +38,42 @@ const addToList = () => {
...
         </div><br></br>
              <div>
-            <button class={`${style.btnCart}`} style={{backgroundColor: 'yellow', color: 'black', }} onClick={addToCart}>Add to Cart</button> 
+            <button  className={`${style.btnCart}`} style={{backgroundColor: 'yellow', color: 'black', }} onClick={addToCart}>Add to Cart</button> 
             </div>
         <div>
-             <button class={`${style.btnBuyNow}`} style={{backgroundColor: 'orange', color: 'black'}} onClick={buyNow}> Buy Now</button> 
+             <button className={`${style.btnBuyNow}`} style={{backgroundColor: 'orange', color: 'black'}} onClick={buyNow}> Buy Now</button> 
         </div><br></br>
         <div id="secure-transaction">
```
Comment (moses369) [inline components/ProductInfo/Cart.js]: Changed class to className

File: components/ProductInfo/ProdImage.js [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+import { ImLink } from "react-icons/im";
 
 import { useSelector, useDispatch } from "react-redux";
-import { changeImage } from "../../redux/features/image-slice";
+import { changeImage } from "../../redux/features/prodInfo-slice";
 
 import style from "../../styles/ProdImage.module.css";
 const ProdImage = () => {
-  const displayImg = useSelector(({ img }) => img.image);
+  const displayImg = useSelector(({ prodInfo }) => prodInfo.image);
```
Comment (moses369) [inline components/ProductInfo/ProdImage.js:13]: Imported Icons for links and updated redux state path

File: components/ProductInfo/ProdImage.js [hunk 3]
```diff
@@ -48,54 +52,62 @@ const ImgBtn = ({ image }) => {
...
-        trackMp[trackMp.length - 2][1] == top - 12
-      ) {
-        setPosition({ display: "block", left, top });
-        setTimeout(() => {}, 1000);
-      }
-    }, 1000);
-
-    if (window.pageX === left - 10 && window.pageY === top) {
-      setPosition({ display: "block", left, top });
-    }
```
Comment (moses369) [inline components/ProductInfo/ProdImage.js:73]: Unessecary code as the hover div is produced by the title attribute

File: components/ProductInfo/ProdImage.js [hunk 4]
```diff
@@ -48,54 +52,62 @@ const ImgBtn = ({ image }) => {
...
+  const [copyText, setCopyText] = useState("Copy Link");
+  const [linksDisplay, setLinksDisplay] = useState({ display: "none" });
+  const copyUrl = useSelector(({ prodInfo }) => prodInfo.pageUrl);
+  const copyLink = () => {
+    navigator.clipboard
+      .writeText(copyUrl)
+      .then(() => setCopyText("Link Copied!"))
+      .then(() => setTimeout(() => setCopyText("Copy Link"), 1000));
```
Comment (moses369) [inline components/ProductInfo/ProdImage.js:62]: Created function to copy url to users clipboard on a click in the in the Copy Link a tag

File: components/ProductInfo/ProdImage.js [hunk 5]
```diff
@@ -48,54 +52,62 @@ const ImgBtn = ({ image }) => {
...
   };
-  const hideTooltip = (e) => {
-    position.display !== "none" && setPosition({ display: "none" });
-    trackMp = [];
```
Comment (moses369) [inline components/ProductInfo/ProdImage.js:77]: Unnecessary code from tooltip being possible through title attr

File: components/ProductInfo/ProdImage.js [hunk 6]
```diff
@@ -48,54 +52,62 @@ const ImgBtn = ({ image }) => {
...
+    tw: "https://www.amazon.com/gp/redirect.html/ref=cm_sw_cl_tw_dp_15W5YPJXG8ZK0V2WN22S?_encoding=UTF8&location=https%3A%2F%2Ftwitter.com%2Fintent%2Ftweet%3Foriginal_referer%3Dhttps%25253A%25252F%25252Fwww.amazon.com%25252Fgp%25252Fproduct%25252FB08QV53WTG%25252Fref%25253Dcm_sw_r_tw_dp_15W5YPJXG8ZK0V2WN22S%25253Fie%25253DUTF8%252526psc%25253D1%26text%3DThe%2520Party%2520Don%27t%2520Start%2520Til%2520l%2520Croc%2520In%2520T-Shirt%2520The%2520Party%2520Don%27t%2520Start%2520Til%2520l%2520Croc%2520...%26related%3Damazondeals%252Camazonmp3%26url%3Dhttps%253A%252F%252Fwww.amazon.com%252Fdp%252FB08QV53WTG%252Fref%253Dcm_sw_r_tw_dp_15W5YPJXG8ZK0V2WN22S%253F_encoding%253DUTF8%2526psc%253D1%26via%3Damazon&token=E5FB53EC80CFED24755EDD959D627E22B8F921F5",
   };
+
```
Comment (moses369) [inline components/ProductInfo/ProdImage.js:71]: Added object of redirect links to not clutter the return statement

File: components/ProductInfo/ProdImage.js [hunk 7]
```diff
@@ -48,54 +52,62 @@ const ImgBtn = ({ image }) => {
...
         aria-label="share"
+        title="Share"
         className={`${style.share}`}
-        onMouseMove={displayTooltip}
-        onMouseLeave={hideTooltip}
+        onClick={() =>
+          linksDisplay.display === "none"
+            ? setLinksDisplay({ display: "flex" })
+            : setLinksDisplay({ display: "none" })
+        }
```
Comment (moses369) [inline components/ProductInfo/ProdImage.js:107]: Linked links to redirect and added icons

File: components/ProductInfo/index.js [hunk 8]
```diff
@@ -1,17 +1,66 @@
...
+import ProdImage from "./ProdImage";
+import Middle from "./middle/Middle";
 
-import style from '../../styles/ProdInfo.module.css'
+import style from "../../styles/ProdInfo.module.css";
 const ProductInfo = () => {
   return (
-    <div className={`${style.container}`}>
+    <>
+      <ProdPath />
```
Comment (moses369) [inline components/ProductInfo/index.js:9]: Added Product path component to div

File: components/ProductInfo/index.js [hunk 9]
```diff
@@ -1,17 +1,66 @@
...
+            <a href={url} className={style.pathLink}>
+              {text}
+            </a>
+            {i !== paths.length - 1 && <span>{">"}</span>}
+          </>
+        );
+      })}
+    </div>
+  );
+};
```
Comment (moses369) [inline components/ProductInfo/index.js:66]: ProductPath component to reflect the path a user could have taken to get here
image

File: redux/features/prodInfo-slice.js [hunk 10]
```diff
@@ -15,5 +16,5 @@ export const imageSlice = createSlice({
   },
 });
 
-export const {changeImage} = imageSlice.actions
-export default imageSlice.reducer
\ No newline at end of file
+export const { changeImage } = prodInfoSlice.actions;
+export default prodInfoSlice.reducer;
```
Comment (moses369) [inline redux/features/prodInfo-slice.js:20]: Edited name to create a more flexible slice and reduce complexity

File: redux/index.js [hunk 11]
```diff
@@ -1,8 +1,8 @@
 import { configureStore } from "@reduxjs/toolkit";
-import imageReducer from "./features/image-slice";
+import prodInfoReducer from "./features/prodInfo-slice";
 export const store = configureStore({
   reducer: {
-    img:imageReducer
+    prodInfo:prodInfoReducer
```
Comment (moses369) [inline redux/index.js:5]: can now be accessed as prodInfo in redux state obj

File: components/ProductInfo/Cart.js (hunk 1 above)
Comment (JacobDWalters) [reply components/ProductInfo/Cart.js]: did you mean to do set classNames without the style. before them?

**Review by CountDown2Extinction — APPROVED** [review]

**Review by ilsaann — APPROVED** [review]

**Pull request merged**